.circleci/config.yml

```diff
@@ -0,0 +1,140 @@
+common_steps: &common_steps
+  steps:
+    - run:
+        name: "Install npm"
+        command: |
+          curl -sL https://deb.nodesource.com/setup_9.x | sudo -E bash -
+          sudo apt-get install -y nodejs
+          mkdir -p ~/.npm-global
+          npm config set prefix $NPM_CONFIG_PREFIX
+    - checkout
+    - run:
+        name: "Note OCAML_VERSION for cache"
+        command: echo $OCAML_VERSION > ../ocaml-version
+    - restore_cache:
+        key: dependency-cache-{{ checksum "../ocaml-version" }}-{{ checksum ".circleci/config.yml" }}-{{ checksum "reason.opam" }}-{{ checksum "rebuild.opam" }}-{{ checksum "rtop.opam" }}
+    - run:
+        name: "Initialize opam"
+        command: |
+          opam init --auto-setup --dot-profile=~/.bash_profile
+          opam remote add ocamlorg https://opam.ocaml.org -p 0 || true
+          opam remote remove default || true
+    - run:
+        name: "Install deps"
+        command: |
+          opam update
+          opam install -y jbuilder
+          opam install -y menhir.20170712
+    - run:
+        name: 'Clean'
+        command: make clean-for-ci
+    - run:
+        name: 'opam pin'
+        command: |
+          opam pin add -y reason .
+          opam pin add -y rtop .
+    - save_cache:
+        key: dependency-cache-{{ checksum "../ocaml-version" }}-{{ checksum ".circleci/config.yml" }}-{{ checksum "reason.opam" }}-{{ checksum "rebuild.opam" }}-{{ checksum "rtop.opam" }}
```

Comment:
Curious if people ever intentionally let the `save_cache` and `restore_cache` diverge.

Reply:
I doubt it--made it use a yaml anchor to prevent people forgetting to update one and not the other.